src/Common/src/Interop/OSX/Interop.CoreFoundation.CFString.cs

```diff
@@ -0,0 +1,82 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Text;
+using Microsoft.Win32.SafeHandles;
+
+internal static partial class Interop
+{
+    internal static partial class CoreFoundation
+    {
+        /// <summary>
+        /// Returns the interior pointer of the cfString if it has the specified encoding.
+        /// If it has the wrong encoding, or if the interior pointer isn't being shared for some reason, returns NULL
+        /// </summary>
+        [DllImport(Libraries.CoreFoundationLibrary)]
+        private static extern IntPtr CFStringGetCStringPtr(
+            SafeCFStringHandle cfString,
+            CFStringBuiltInEncodings encoding);
+
+        [DllImport(Libraries.CoreFoundationLibrary)]
+        private static extern SafeCFDataHandle CFStringCreateExternalRepresentation(
+            IntPtr alloc,
+            SafeCFStringHandle theString,
+            CFStringBuiltInEncodings encoding,
+            byte lossByte);
+
+        internal static string CFStringToString(SafeCFStringHandle cfString)
+        {
+            Debug.Assert(cfString != null);
+            Debug.Assert(!cfString.IsInvalid);
+            Debug.Assert(!cfString.IsClosed);
+
+            IntPtr interiorPointer = CFStringGetCStringPtr(
+                cfString,
+                CFStringBuiltInEncodings.kCFStringEncodingUTF8);
+
+            if (interiorPointer != IntPtr.Zero)
+            {
+                return Marshal.PtrToStringUTF8(interiorPointer);
+            }
+
+            SafeCFDataHandle cfData = CFStringCreateExternalRepresentation(
+                IntPtr.Zero,
+                cfString,
+                CFStringBuiltInEncodings.kCFStringEncodingUTF8,
+                0);
+
+            byte[] strBytes;
+
+            using (cfData)
+            {
+                strBytes = CFGetData(cfData);
+            }
+
+            return Encoding.UTF8.GetString(strBytes);
```

Comment:
This is one of the places where it seems like we could go directly from the SafeCFDataHandle to a string, without going through an intermediate byte[].